src/app/static/src/app/components/downloadResults/UploadModal.vue

```diff
@@ -100,10 +126,58 @@
                 "uploadFilesToADR"
             ),
             confirmUpload() {
-                const uploadFilesPayload: UploadFile[] = []
-                this.uploadFilesToAdr.forEach(value => uploadFilesPayload.push(this.uploadFiles[value]))
-                this.uploadFilesToADRAction(uploadFilesPayload);
-                this.$emit("close")
+                this.uploadFilesToAdr.forEach(value => this.uploadFilesPayload.push(this.uploadFiles[value]));
+                const readyForUpload = this.prepareFilesForUpload();
+
+                if (readyForUpload) {
+                    this.sendUploadFilesToADR();
+                }
+            },
+            sendUploadFilesToADR() {
+                this.uploadFilesToADRAction(this.uploadFilesPayload);
+                this.uploadFilesPayload = [];
+
+                this.$emit("close");
+            },
+            prepareFilesForUpload() {
+                const {summary, spectrum} = this.findSelectedUploadFiles();
+                if (summary) {
+                    this.getSummaryDownload();
+                }
+                if (spectrum) {
+                    this.getSpectrumDownload();
+                }
+
+                return this.downloadIsReady();
+            },
+            findSelectedUploadFiles() {
+                const summary = this.uploadFilesPayload.find(upload => upload.resourceType === this.outputSummary);
+                const spectrum = this.uploadFilesPayload.find(upload => upload.resourceType === this.outputSpectrum);
+
+                return {summary, spectrum}
+            },
+            downloadIsReady() {
+                const {summary, spectrum} = this.findSelectedUploadFiles();
+
+                if (summary && spectrum) {
+                    return !!this.summary.complete && !!this.spectrum.complete;
+                } else {
+                    if (summary && !spectrum) {
+                        return !!this.summary.complete;
+                    } else {
+                        return !!this.spectrum.complete;
+                    }
+                }
```

Comment:
Could you please respond to this suggestion, either by accepting it or saying why you disagree?

Reply:
If I have got it a bit wrong, would it still be possible to rework your code to be more in this style, as both Mark and I feel that it makes the logic more visible than multiple `if` statements. 
Which case are you thinking of that this would not work for? Is it the case where neither summary nor spectrum are checked ? That should return false because downloadIsReady should be false if neither are ticked? Could add a clause to check at least one of the two is true?:
`return (summary || spectrum) && (!summary || !!this.summary.complete) && (!spectrum || !!this.spectrum.complete);`